hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/AsyncMapWriter.java

```diff
@@ -230,15 +234,17 @@ private boolean invokeOnCluster(List<PartitionOpBuilder> opBuilders,
                     }
 
                     // retry once
-                    final MapEntries[] entries = failedEntries.toArray(new MapEntries[failedEntries.size()]);
+                    final MapEntries[] entries = failedEntries.toArray(new MapEntries[0]);
                     final int[] partitions = toIntArray(failedPartitions);
                     final Throwable originalErr = error;
                     executionService.schedule(() -> {
                         try {
-                            if (!tryRetry(partitions, entries, completionFuture)) {
+                            if (!tryRetry(partitions, entries, doneLatch, completionFuture)) {
                                 completionFuture.completeExceptionally(originalErr);
+                                return;
                             }
-
+                            int value = doneLatch.decrementAndGet();
+                            assert value > 0 : "value=" + value;
```

Comment:
couldn't value be 0 here if the retry completes before the assert?

Reply:
yes, that's true. But then I also need to do the same action as the other place if the count gets to 0. I'll fix it.